pkg/odo/cli/dev/dev.go

```diff
@@ -0,0 +1,131 @@
+package dev
+
+import (
+	"fmt"
+	dfutil "github.com/devfile/library/pkg/util"
+	"github.com/redhat-developer/odo/pkg/devfile/adapters/kubernetes"
+	"github.com/redhat-developer/odo/pkg/envinfo"
+	"github.com/redhat-developer/odo/pkg/odo/cli/component"
+	"github.com/redhat-developer/odo/pkg/odo/cmdline"
+	"github.com/redhat-developer/odo/pkg/odo/genericclioptions"
+	"github.com/redhat-developer/odo/pkg/odo/genericclioptions/clientset"
+	odoutil "github.com/redhat-developer/odo/pkg/odo/util"
+	"github.com/redhat-developer/odo/pkg/util"
+	"github.com/spf13/cobra"
+	"k8s.io/kubectl/pkg/util/templates"
+	"os"
+	"path/filepath"
+)
+
+// RecommendedCommandName is the recommended command name
+const RecommendedCommandName = "dev"
+
+type DevOptions struct {
+	// Context
+	*genericclioptions.Context
+
+	// Clients
+	clientset *clientset.Clientset
+
+	// Variables
+	ignorePaths []string
+}
+
+func NewDevOptions() *DevOptions {
+	return &DevOptions{}
+}
+
+var devExample = templates.Examples(`
+	# Deploy component to the development cluster
+	%[1]s
+`)
+
+func (o *DevOptions) SetClientset(clientset *clientset.Clientset) {
+	o.clientset = clientset
+}
+
+func (o *DevOptions) Complete(cmdline cmdline.Cmdline, args []string) error {
+	var err error
+
+	o.Context, err = genericclioptions.New(genericclioptions.NewCreateParameters(cmdline).NeedDevfile(""))
+	if err != nil {
+		return fmt.Errorf("unable to create context: %v", err)
+	}
+
+	devfileExists := util.CheckPathExists(o.Context.GetDevfilePath())
+	if !devfileExists {
+		return fmt.Errorf("the current directory doesn't contain a devfile")
+	}
+
```

Comment:
I think this is already tested when you call `genericlioptions.New()` with `NeedDevfile()`

Reply:
Thanks for the catch. Removed.